src/common.ts

```diff
@@ -0,0 +1,28 @@
+import { User, UserService } from './scripts/services/user-service';
+import * as TE from 'fp-ts/lib/TaskEither';
+import { flow, pipe } from 'fp-ts/lib/function';
+import * as O from 'fp-ts/lib/Option';
+
+/**
+ * Retrieves a user's details from Okta
+ * @param service - the service to use to retrieve the user.
+ * @param userId - the id of the user to retrieve.
+ * @returns a TaskEither that resolves to the user.
+ */
+export const retrieveUser = (
```

Comment:
**issue:** a file named `common.ts` is a bit of a code smell. It implies that there is no coherency to the content of it. These tend to become a bit of a dumping ground over time.

I would *rather* that do this in the relevant scripts:

```
pipe(
    ...
    service.getUser,
    TE.chain(TE.fromOption(`User [${userId}] does not exist, can not activate.`)),
    ...
)
```

Noting that we can tailor the `onLeft` to describe the impact of the user not existing.

Reply:
removed in commit: 354384fceb62181d04fada17fee25e72c7d45e28